distinguish between rootpath and valid subpaths fix #14

MartinLichtblau opened the pull request:
distinguish between rootpath and valid subpaths
fix #14

MartinLichtblau committed:
distinguish between rootpath and valid subpaths
fix #14

MartinLichtblau committed:
distinguish between rootpath and valid subpaths
fix #14

vyo commented:
Hmm, looks solid. You may merge if you want to. Coverage looks good, too:

No global variable leaks detected
Coverage: 86.07% (50/359)

Issue specific comments are in #14.

vyo commented:
Just as an aside, leave a blank line after your first commit line - that way only the first line shows up as a sort of headline, makes it easer on the eyes.
Perhaps we should agree on some uniform commit style? Yours is really close to mine, I'd just like to have something uniform.

vyo commented:
Hmm, would you mind changing rootpath to basepath? Minor, but root makes me think of something more fundamental than what we're specifying here.

MartinLichtblau commented:
Yeah agree, that was the first name I gave it. Don't know why I changed it.
Will do a PR tomorrow.

MartinLichtblau commented:
- Didn't implement any tests for that piece, probably should!?
- Asides are good, thx!
- We could agree use the [style of the jhipster project](https://github.com/jhipster/generator-jhipster/blob/79a4abfc550ebd5cedd9c648d0f13b59816cefd5/CONTRIBUTING.md#-git-commit-guidelines)?

MartinLichtblau committed:
allow multiple valid paths

descriminate between basepath and valid subpaths and create valid paths like "basepath/core/subpath"

fix #14

vyo commented:
Would be nice. Take a look at how I mocked the solr client, might help. I'd
suggest mocking the solr server and creating at least a positive and a
negative test. Not sure if we should mock or test the proxy, too.
Suggestions @MartinLichtblau?

On Tue, 7 Jul 2015 23:33 MartinLichtblau notifications@github.com wrote:

> - Didn't implement any tests for that piece, probably should!?
> - Asides are good, thx!
> - We could agree use the style of the jhipster project
>   https://github.com/jhipster/generator-jhipster/blob/79a4abfc550ebd5cedd9c648d0f13b59816cefd5/CONTRIBUTING.md#-git-commit-guidelines
>   ?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/cpsgroup/sarapis/pull/15#issuecomment-119349059.
